scipy/special/tests/test_basic.py

```diff
@@ -1319,8 +1319,8 @@ def test_comb(self):
         expected = 100891344545564193334812497256
         assert_equal(special.comb(100, 50, exact=True), expected)
 
-        assert_raises(ValueError, special.comb, 4.5, 3, True)
-        assert_raises(ValueError, special.comb, 4, 3.5, True)
+        assert_raises(DeprecationWarning, special.comb, 4.5, 3, True)
+        assert_raises(DeprecationWarning, special.comb, 4, 3.5, True)
```

Comment:
You'll need to express this differently; pytest.raises [deals with](https://docs.pytest.org/en/7.0.x/reference/reference.html#pytest.raises) exceptions, not warnings.

Would also be good to write this using the contextmanager-form (twice) of pytest.warns (see [here](https://docs.pytest.org/en/7.0.x/reference/reference.html#pytest-warns)).